packages/flutter_tools/test/general.shard/android/deferred_components_setup_validator_test.dart

```diff
@@ -0,0 +1,911 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/android/deferred_components_setup_validator.dart';
+import 'package:flutter_tools/src/base/deferred_component.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/build_system/targets/common.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+void main() {
+  final FileSystem fileSystem = MemoryFileSystem.test();
+  final BufferLogger logger = BufferLogger.test();
+
+  Environment createEnvironment() {
+    final Map<String, String> defines = <String, String>{ kSplitAot: 'true' };
+    final Environment result = Environment(
+      outputDir: fileSystem.directory('/output'),
+      buildDir: fileSystem.directory('/build'),
+      projectDir: fileSystem.directory('/project'),
+      defines: defines,
+      inputs: <String, String>{},
+      cacheDir: fileSystem.directory('/cache'),
+      flutterRootDir: fileSystem.directory('/flutter_root'),
+      artifacts: globals.artifacts,
+      fileSystem: fileSystem,
+      logger: logger,
+      processManager: globals.processManager,
+      engineVersion: 'invalidEngineVersion',
+    );
+    return result;
+  }
+  final Environment env = createEnvironment();
+
+  test('No checks passes', () async {
+    logger.clear();
```

Comment:
This is a dangerous pattern because forgetting to clear might lead to future tests passing inadvertently

Reply:
I'll use the setup method to avoid this.